projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -212,6 +216,9 @@ export class IgxOverlayService implements OnDestroy {
         }
 
         const eventArgs = { id, componentRef: info.componentRef, cancel: false };
+        if (event) {
+            Object.assign(eventArgs, { event });
+        }
```

Comment:
you can directly do it like this:
const eventArgs = { id, componentRef: info.componentRef, cancel: false, event };

no need to check if there is event and object assign it after.